### DISCUSSION TITLE
APK packaging issue

### DISCUSSION THREAD

**Discussion by Maskedviper** [flet-dev/flet#2657] [Q&A]
### Question

![WhatsApp Image 2024-02-12 at 12 12 55 PM (1)](https://github.com/flet-dev/flet/assets/10101674/208751aa-3a14-4e7b-a8ca-411e28e9fa93)

![WhatsApp Image 2024-02-15 at 12 13 49 PM](https://github.com/flet-dev/flet/assets/10101674/30642519-fb51-4e7f-9597-5acb87df13a6)

When we add the libraries one at a time they modulenotfounderror Is removed for the previous library not included but when we include all libraries in one go its taking hours for the packaging process and it still doesn't end up packaging. Like when I include flet,requests and bs4 to the dependencies the apk is being created. As soon as I install pandas into the virtual environment it stalls the packaging process. 

The Pandas library seems to be giving the issue as without this library the APK is able to package. Without it, it goes to the end and it shows apk success. I also installed the latest version of pandas into the virtual environment

Could it be due to lack of memory (RAM) causing it to crash?

Thanks for the help!

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by FeodorFitsner** [comment]
https://flet.dev/docs/guides/python/packaging-app-for-distribution#native-python-packages

`Numpy` (and maybe some others) is a native package.

**Reply by Maskedviper** [reply]
Hi,

After reading through that document, I just had a few questions:

**Run p4a with --requirements including your custom Python libraries separated with comma, like numpy in the following example:

**"p4a create --requirements numpy --arch arm64-v8a --arch armeabi-v7a --arch x86_64 --sdk-dir $ANDROID_SDK_ROOT --ndk-dir $ANDROID_SDK_ROOT/ndk/$NDK_VERSION --dist-name mydist"****

Like for this step, how do we correlate it to our project folder in our directories we are using for our project.

**$HOME/.python-for-android/dists/mydist**
**export SERIOUS_PYTHON_P4A_DIST=$HOME/.python-for-android/dists/mydist**

Like even for these steps, how do we correlate the directories to our own project?

Thank you for your help!

**Reply by FeodorFitsner** [reply]
What do you mean by correlate?

**Reply by Maskedviper** [reply]
So for this p4a create --requirements numpy --arch arm64-v8a --arch armeabi-v7a --arch x86_64 --sdk-dir $ANDROID_SDK_ROOT --ndk-dir $ANDROID_SDK_ROOT/ndk/$NDK_VERSION --dist-name mydist"

I am just confused about the directory names. Like what is the purpose of this command or is this just an example?

**Reply by FeodorFitsner** [reply]
It's an example. You should put the list of your Python packages instead of `numpy`.

**Reply by Maskedviper** [reply]
Thank you! I will try that
